SmartDeviceLink/private/SDLTextAndGraphicManager.m

```diff
@@ -150,21 +151,29 @@ - (void)updateWithCompletionHandler:(nullable SDLTextAndGraphicUpdateCompletionH
 
     if (self.isDirty) {
         self.isDirty = NO;
-        [self sdl_updateWithCompletionHandler:handler];
+        [self sdl_updateAndCancelPreviousOperations:YES completionHandler:handler];
     }
 }
 
-- (void)sdl_updateWithCompletionHandler:(nullable SDLTextAndGraphicUpdateCompletionHandler)handler {
+- (void)sdl_updateAndCancelPreviousOperations:(BOOL)supersedePreviousOperations completionHandler:(nullable SDLTextAndGraphicUpdateCompletionHandler)handler {
     SDLLogD(@"Updating text and graphics");
-    if (self.transactionQueue.operationCount > 0) {
+    if (self.transactionQueue.operationCount > 0 && supersedePreviousOperations) {
```

Comment:
They're not unrelated though because they can happen in the same update. If you're on a 6.0+ system it'll actually happen in one request all together instead of two separate ones which is greatly preferable. 

Separating them in some cases like this and not in others would be a hell of edge-case spaghetti code. Most of the time we want to combine the updates. Otherwise if there are updates like:

```
screenManager.textField1 = "Blah"
screenManager.textField2 = "Blah"
screenManager.textField3 = "Blah"
screenManager.textField4 = "Blah"
```

It will all happen with separate updates instead of one or two depending on timing.

We could keep the handler around until another update with a handler has superseded that one. It would work in some cases, but (1) it makes things significantly more complicated, and (2) it may fire for an unrelated update. For example:

```
screenManager.beginUpdates()
screenManager.textField1 = "Blah"
screenManager.textField2 = "Blah"
screenManager.endUpdates { error in
  // Do something
}

screenManager.beginUpdates()
screenManager.textField1 = "Blah"
screenManager.textField2 = "Blah"
screenManager.endUpdates()
```

The update handler may fire due to the second update instead of the first because the first gets superseded by the second if the first didn't start by the time the second was entered.

Last, your case is relatively unlikely, or it should be. Most people shouldn't change the layout _then_ batch update. It should include the layout change in the batch update. Basically what you're doing is breaking best practice, so you get a successful, though undesirable, result.

All of this gets very confusing, I know, but I think this is very difficult to fix, and any given fix would be far too complicated to be worthwhile. Believe me, Julian and I talked this over 100 different ways. I have added a user-readable description  to the error and documentation to the methods that should help the developer if they encounter the superseded error.

Reply:
Ok. It is much clearer what is happening now with the new docs and better warning message.